content/submit-applications/docs/spark/pyspark.md

```diff
@@ -683,9 +687,151 @@ Upload `numpy_environment.tar.gz` to a s3 location that is mapped to a FSx for L
 
 #### Bundled as virtual env  
 
-**This will not work with spark on kubernetes**.This feature only works with YARN - cluster mode
+!!! warning
+    **This will not work with spark on kubernetes**
+
+This feature only works with YARN - cluster mode
 In this implementation for YARN - the dependencies will be installed from the repository for every driver and executor. This might not be a more scalable model as per [SPARK-25433](https://issues.apache.org/jira/browse/SPARK-25433). Recommended solution is to pass in the dependencies as PEX file.
 
+#### Custom docker image
+See the details in the [official documentation](https://docs.aws.amazon.com/emr/latest/EMR-on-EKS-DevelopmentGuide/docker-custom-images.html).  
+
+**Dockerfile**
+```dockerfile
+FROM 107292555468.dkr.ecr.eu-central-1.amazonaws.com/spark/emr-6.3.0
+USER root
+RUN pip3 install boto3
+USER hadoop:hadoop
+```
+
+### Python code with java dependencies  
+
+####**List of packages**
+
+!!! warning
+    **This will not work with spark on kubernetes**
+
+This feature only works with YARN - cluster mode
```

Comment:
This does seem to work if I use a `sparkSubmitParameters` value of `"--packages org.apache.spark:spark-sql-kafka-0-10_2.12:3.1.2"` - I didn't test to see if I could import kafka, but it did download the dependencies on the driver:


```
Ivy Default Cache set to: /home/hadoop/.ivy2/cache
The jars for the packages stored in: /home/hadoop/.ivy2/jars
:: loading settings :: url = jar:file:/usr/lib/spark/jars/ivy-2.4.0.jar!/org/apache/ivy/core/settings/ivysettings.xml
org.apache.spark#spark-sql-kafka-0-10_2.12 added as a dependency
:: resolving dependencies :: org.apache.spark#spark-submit-parent-160bca5d-34a1-426f-a39f-d7cb17cda966;1.0
        confs: [default]
        found org.apache.spark#spark-sql-kafka-0-10_2.12;3.1.2 in central
        found org.apache.spark#spark-token-provider-kafka-0-10_2.12;3.1.2 in central
        found org.apache.kafka#kafka-clients;2.6.0 in central
        found com.github.luben#zstd-jni;1.4.8-1 in central
        found org.lz4#lz4-java;1.7.1 in central
        found org.xerial.snappy#snappy-java;1.1.8.2 in central
        found org.slf4j#slf4j-api;1.7.30 in central
        found org.spark-project.spark#unused;1.0.0 in central
        found org.apache.commons#commons-pool2;2.6.2 in central
downloading https://repo1.maven.org/maven2/org/apache/spark/spark-sql-kafka-0-10_2.12/3.1.2/spark-sql-kafka-0-10_2.12-3.1.2.jar ...
        [SUCCESSFUL ] org.apache.spark#spark-sql-kafka-0-10_2.12;3.1.2!spark-sql-kafka-0-10_2.12.jar (15ms)
```

Reply:
Yes the download will take place only on the driver but none of the executor will download the file and the job will fail. Executors only will download the files if URL is provided

Reply:
@dacort This method was tested and was true on EMR 6.3.0. I don't know it is true for newer version. Since I changed job and I don't have access to EMR on EKS environment.

My generic advice to avoid all the import use custom docker image

Reply:
@andormarkus ok thanks for the quick reply! I'll try to double-check tomorrow, appreciate your contribution!

Reply:
OK yep, since EMR on EKS uses `--deploy-mode client`, the executors won't pull down the packages themselves.  At least I think that's the reason. :) 

So yes, generic advice would be to use custom docker image. I have an example I started as a PR and can add it here.